### ISSUE TITLE
Unable to access USB Ports

### ISSUE THREAD

**Issue by ChaosUntamed** [dnschneid/crouton#4877]
```
Please paste the output of the following command here: sudo edit-chroot -all
```

#### Please describe your issue:
Hello everyone,

I installed Ubuntu Focal Fossa 20.04 on my old chromebook using [this site](https://github.com/dnschneid/crouton/issues/4265#issuecomment-689346174) as a guide. Everything seems to have gone just fine. But now, when I plug something into the USB ports, it pops up in the File Directory. But it only lasts a second or two and then disappears. 

This only happens in the chroot environment. In the ChromeOS environment, the USB ports work fine. I did not provide the information above because the command returned an error that said it was not found. 

Any help would be greatly appreciated. I need the USB ports to work to connect my 3d printer to the chromebook.

**Comment by TomTravis** [comment]
hello i assume your using the cups printer driver 
yes to ports are closed in change root 

my suggestion is to print spool in lpr to a file and send it via chome lpr not cups

**Comment by ChaosUntamed** [comment]
> hello i assume your using the cups printer driver 
> yes to ports are closed in change root 
> 
> my suggestion is to print spool in lpr to a file and send it via chome lpr not cups 
> 
> or open the port
> port 631
> CUPS uses port 631 (TCP and UDP), which is the standard IPP port, and optionally on port 515 by inetd, launchd, the Solaris Service Management Facility, or xinetd which use the cups-lpd helper program to support LPD printing.

Wow. That was way more complicated than I can even comprehend. 

What I intended to do was install OctoPrint in the Linux environment and connect the printer via a micro USB (printer) to chromebook (usb).

But I need the USB ports to be accessible from the Linux environment. Would opening Port 631 do that? And how would I do so?

**Comment by TomTravis** [comment]
it make cups happy
read this
https://github.com/dnschneid/crouton/issues/143

**Comment by TomTravis** [comment]
octo print has a tcipip server octoserver use it

**Comment by ChaosUntamed** [comment]
> octo print has a tcipip server octoserver use it

Will this solve my usb port problem?

**Comment by TomTravis** [comment]
yes i have no idea you have to test it

**Comment by TomTravis** [comment]
> 



> > yes i have no idea you have to test it
> 
> I would love to. If I even knew what "it" is. I know very little about any of this. I've just been trying to figure out how to use OctoPrint on this chromebook I bought and it has led me down a very long, dark tunnel that I know nothing about...
> 
> I'm sorry if I'm wasting your time. I just don't understand any of this.

usb print hub server tcp/ip
https://www.amazon.com/U-S-ROBOTICS-Server-APLTLK-USR7500/dp/B0002VO7HE

this is you plug and play

**Comment by ChaosUntamed** [comment]
How will that help when the chromebook won't read from the USB ports in the Ubuntu environment? Wouldn't that have the same problem?

**Comment by TomTravis** [comment]
forget USB on the CB 
octoprint serve is a tcpip serve the connect to the usb print server via the wireless network no usb ubuntu needes

**Comment by ChaosUntamed** [comment]
I would rather see if I can get my USB ports to work on the CB. Maybe that sounds dumb and is overly complicated, but I've already invested money and a lot of effort in this route. 

That said, if I can't get the USB ports to work, I'll see if I can't figure out how to use the Print Server that you posted as an alternative. Thank you for explaining things to me and taking the time to work with me on this.

**Comment by TomTravis** [comment]
> I would rather see if I can get my USB ports to work on the CB. Maybe that sounds dumb and is overly complicated, but I've already invested money and a lot of effort in this route.
> 
> That said, if I can't get the USB ports to work, I'll see if I can't figure out how to use the Print Server that you posted as an alternative. Thank you for explaining things to me and taking the time to work with me on this.
join the club so time is money play with USB drivers in Octoprint that does not use them 
of 39$ it works

**Comment by ChaosUntamed** [comment]
That's a fair point. If I got that device. How would I use it?

**Comment by TomTravis** [comment]
> That's a fair point. If I got that device. How would I use it?

read octoserver port 8080 send to print server simple  config

**Comment by ChaosUntamed** [comment]
I've had a long day and apparently my mind just isn't working with me. I'm utterly confused.
From my limited understanding currently, I get the device and plug it into my 3d printer. Then load octoprint up on my CB and access the wifi signal via the Ubuntu terminal using the commands specified. 

I don't know that that's right, but I'm gonna try to sleep on it and read through this again. Maybe it will click for me. Again, thank you for all your help.

**Comment by TomTravis** [comment]
> I've had a long day and apparently my mind just isn't working with me. I'm utterly confused. From my limited understanding currently, I get the device and plug it into my 3d printer. Then load octoprint up on my CB and access the wifi signal via the Ubuntu terminal using the commands specified. No idea what the "build and install" part means.
> 
> I don't know that that's right, but I'm gonna try to sleep on it and read through this again. Maybe it will click for me. Again, thank you for all your help.

https://wiki.debian.org/CUPSDriverlessPrinting
some reading

**Comment by ChaosUntamed** [comment]
> hello i assume your using the cups printer driver 
> yes to ports are closed in change root 
> https://github.com/dnschneid/crouton/issues/143
> my suggestion is to print spool in lpr to a file and send it via chome lpr not cups 
> 
> or open the port
> port 631
> CUPS uses port 631 (TCP and UDP), which is the standard IPP port, and optionally on port 515 by inetd, launchd, the Solaris Service Management Facility, or xinetd which use the cups-lpd helper program to support LPD printing.

Reading back through all this. I'd like to try opening port 631 and maybe 515. 

CUPS is installed. How do I open a port using it? Specifically, what commands?

**Comment by TomTravis** [comment]
yes okay you like tcp then join the team and we go from there you have a invite and task 

the command are not cli and you submit a table with command kinda like a print job
man ip-tables

**Comment by ChaosUntamed** [comment]
Join the team? What exactly do you want me to do? I don't know enough about this stuff to even understand a single thing you said... 

I'd love to help, but I'm drowning in my own inadequacy here. Very much out of my element.

**Comment by TomTravis** [comment]
you have a invite yes, systemd chroot team
[2ip.txt](https://github.com/dnschneid/crouton/files/9994274/2ip.txt)


here is a IPtable

**Comment by supechicken** [comment]
What is the output of `sudo lsusb` command (inside chroot) after you connected the 3d printer?

Btw better to not listen to the guy above, I don't think he understand what he saying. (I guess all his comments were directly copied from results in google)

**Comment by ChaosUntamed** [comment]
> What is the output of `sudo lsusb` command (inside chroot) after you connected the 3d printer?
> 
> Btw better to not listen to the guy above, I don't think he understand what he said. (it seems that all of his comments were directly copied from results in google)

![20221111_225200.jpg](https://user-images.githubusercontent.com/12370189/201461602-fb965043-3829-46fa-a06d-bb5d8e47a838.jpg)

This is what I get from that command.

**Comment by ChaosUntamed** [comment]
> What is the output of `sudo lsusb` command (inside chroot) after you connected the 3d printer?
> 
> Btw better to not listen to the guy above, I don't think he understand what he said. (it seems that all of his comments were directly copied from results in google)

![20221111_225521.jpg](https://user-images.githubusercontent.com/12370189/201461716-3a03d6bf-7c95-468f-b504-314b856985e3.jpg)

This is what I get running that command with the printer connected.

**Comment by TomTravis** [comment]
tcp is the way octoprint is not usb

(bull)z@localhost:~$ sudo apt install usbutils
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following NEW packages will be installed:
  usbutils
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 75.9 kB of archives.
After this operation, 325 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian bullseye/main arm64 usbutils arm64 1:013-3 [75.9 kB]
Fetched 75.9 kB in 0s (239 kB/s)    
Selecting previously unselected package usbutils.
(Reading database ... 162659 files and directories currently installed.)
Preparing to unpack .../usbutils_1%3a013-3_arm64.deb ...
Unpacking usbutils (1:013-3) ...
Setting up usbutils (1:013-3) ...
Processing triggers for man-db (2.9.4-2) ...
(bull)z@localhost:~$ lsusb -v
[lsusb.txt](https://github.com/dnschneid/crouton/files/9994396/lsusb.txt)

**Comment by TomTravis** [comment]
![Screenshot_2022-11-12_03-30-37](https://user-images.githubusercontent.com/46386261/201471970-a6c1bb60-ed76-45c3-817c-d25285d858d5.png)

**Comment by ChaosUntamed** [comment]
> tcp is the way octoprint is not usb

Even beyond this,  I'd like to get my usb ports working just to use other things that connect via USB. Such as memory drives, USB cd drives, and such.

**Comment by TomTravis** [comment]
well run in /usr/local/bin
yes my camera does not automnt like usb memory, so octoprint does not use usb also you have to make it us this

@superchicken provided and wonderful solution lsusb, this is why i use the shared drive with chromeOS

**Comment by TomTravis** [comment]
ok i believe the USB ports devices and not shareable between chromeOS and chroot linux 
if one has cmd the other does not

the idea is to get get chrome to stop Automount devices if you wish to use them in chroot
i do not use USB ports that much no need i dont have a cd drive or usb printer only mount phones and use the share folder with chromeOS

**Comment by TomTravis** [comment]
> What is the output of `sudo lsusb` command (inside chroot) after you connected the 3d printer?
> 
> Btw better to not listen to the guy above, I don't think he understand what he said. (it seems that all of his comments were directly copied from results in google)

you do not have any idea, what have you done ***nothing***
